core/tests/behavior/write.rs

```diff
@@ -580,6 +581,36 @@ pub async fn test_read_with_special_chars(op: Operator) -> Result<()> {
     Ok(())
 }
 
+pub async fn test_read_with_override_content_disposition(op: Operator) -> Result<()> {
+    if op.info().can_read_with_override_content_disposition() {
+        let path = uuid::Uuid::new_v4().to_string();
+        let (content, _) = gen_bytes();
+
+        op.write(&path, content.clone())
+            .await
+            .expect("write must succeed");
+
+        let target_content_disposition = "attachment; filename=foo.txt";
+
+        let op_read = OpRead::default();
+        op_read.with_override_content_disposition(target_content_disposition.clone());
+
+        let mut reader = op.reader_with(&path, op_read)
+            .await
+            .expect("reader must succeed");
+
+        let mut buf = Vec::new();
+        reader.read_to_end(&mut buf).await?;
+
+        assert_eq!(buf, content);
+        assert_eq!(reader.content_disposition(), Some("attachment; filename=foo.txt"));
```

Comment:
There is no such method to get metadata for now.

Reply:
Nice catch, we didn't expose metadata on read yet. And we aren't ready to expose `Metadata` on reader directly. Need some more thinking.

Reply:
So it is not quite easy to complete the test now.

Reply:
Please note that read operation in OpenDAL is far more complex that a simple http request.

Reply:
> So it is not quite easy to complete the test now.

Yes, we can' test `override_content_disposition` directly. But we can test it via presign (which is the real use case for `override_content_disposition`).

We can presign a read with override_content_disposition and check if the returning response containts the correct `content_disposition` header.

Reply:
Great idea.